Tests/Handler/RemoveHandlerTest.php

```diff
@@ -0,0 +1,223 @@
+
+        $queue = $this->handler->getQueue();
+        $this->assertEquals(count($queue), 0);
+
+        $this->handler->addToQueue($object1, self::FILE_FIELD_ONE);
+        $this->assertEquals(count($queue), 1);
+        $this->assertEquals(
+            $queue[$object1]['fieldNames'],
+            [
+                self::FILE_FIELD_ONE,
+            ]
+        );
+
+        $this->handler->addToQueue($object1, self::FILE_FIELD_TWO);
+        $this->assertEquals(count($queue), 1);
+        $this->assertEquals(
+            $queue[$object1]['fieldNames'],
+            [
+                self::FILE_FIELD_TWO,
+                self::FILE_FIELD_ONE,
+            ]
+        );
+
+        $this->handler->addToQueue($object2, self::FILE_FIELD_ONE);
+        $this->assertEquals(count($queue), 2);
+        $this->assertEquals(
+            $queue[$object2]['fieldNames'],
+            [
+                self::FILE_FIELD_ONE,
+            ]
+        );
+
+        $this->handler->addToQueue($object2, self::FILE_FIELD_TWO);
+        $this->assertEquals(count($queue), 2);
+        $this->assertEquals(
+            $queue[$object2]['fieldNames'],
+            [
+                self::FILE_FIELD_TWO,
+                self::FILE_FIELD_ONE,
+            ]
+        );
+    }
+
+    public function testRemoveFilesInQueue(): void
+    {
+        $object1 = new Article();
+        $object2 = new Article();
+
+        $mapping11 = $this->getPropertyMappingMock();
+        $mapping12 = $this->getPropertyMappingMock();
+        $mapping21 = $this->getPropertyMappingMock();
+        $mapping22 = $this->getPropertyMappingMock();
+
+        // factory calls during addToQueue
+        $this->factoryExpectsFromField(0, $object1, self::FILE_FIELD_ONE, $mapping11);
+        $this->factoryExpectsFromField(1, $object1, self::FILE_FIELD_TWO, $mapping12);
+        $this->factoryExpectsFromField(2, $object2, self::FILE_FIELD_ONE, $mapping21);
+        $this->factoryExpectsFromField(3, $object2, self::FILE_FIELD_TWO, $mapping21);
+
+        // factory calls during remove
+        $this->factoryExpectsFromField(4, $object1, self::FILE_FIELD_TWO, $mapping12);
+        $this->factoryExpectsFromField(5, $object1, self::FILE_FIELD_ONE, $mapping11);
+        $this->factoryExpectsFromField(6, $object2, self::FILE_FIELD_TWO, $mapping22);
+        $this->factoryExpectsFromField(7, $object2, self::FILE_FIELD_ONE, $mapping21);
+
+        $this->handler->addToQueue($object1, self::FILE_FIELD_ONE);
+        $this->handler->addToQueue($object1, self::FILE_FIELD_TWO);
+        $this->handler->addToQueue($object2, self::FILE_FIELD_ONE);
+        $this->handler->addToQueue($object2, self::FILE_FIELD_TWO);
+
+        $objects = $this->handler->removeFilesInQueue();
+
+        $this->assertEquals(count($objects), 2);
+        $this->assertTrue($objects[0] === $object1);
+        $this->assertTrue($objects[1] === $object2);
+    }
+
+    protected function getStorageMock()
```

Comment:
I guess that this test class is not meant to be extended. Protected methods should be private instead